### PR TITLE
Add label selector for GetPods API

### PR DESCRIPTION
**Pull request by disrani-px**

### CODE REVIEW

File: k8s/k8s.go [hunk 1]
```diff
@@ -3039,14 +3040,14 @@ func (k *k8sOps) ValidateClusterPair(name string, namespace string, timeout, ret
 			clusterPair.Status.StorageStatus == v1alpha1.ClusterPairStatusError {
 			return "", true, &ErrFailedToValidateCustomSpec{
 				Name:  name,
-				Cause: fmt.Sprintf("Storage Status %v \t Schedular Status %v", clusterPair.Status.StorageStatus, clusterPair.Status.SchedulerStatus),
+				Cause: fmt.Sprintf("Storage Status: %v \t Schedular Status: %v", clusterPair.Status.StorageStatus, clusterPair.Status.SchedulerStatus),
```
Comment (piyush-nimbalkar) [inline k8s/k8s.go]: nit:  *scheduler

File: k8s/k8s.go [hunk 2]
```diff
@@ -3039,14 +3040,14 @@ func (k *k8sOps) ValidateClusterPair(name string, namespace string, timeout, ret
...
 				Type:  clusterPair,
 			}
 		}
 
 		return "", true, &ErrFailedToValidateCustomSpec{
 			Name:  name,
-			Cause: fmt.Sprintf("Storage Status %v \t Schedular Status %v", clusterPair.Status.StorageStatus, clusterPair.Status.SchedulerStatus),
+			Cause: fmt.Sprintf("Storage Status: %v \t Schedular Status: %v", clusterPair.Status.StorageStatus, clusterPair.Status.SchedulerStatus),
```
Comment (piyush-nimbalkar) [inline k8s/k8s.go]: nit: *scheduler

File: k8s/k8s.go [hunk 3]
```diff
@@ -1982,8 +1982,10 @@ func (k *k8sOps) CreatePod(pod *v1.Pod) (*v1.Pod, error) {
 	return k.client.Core().Pods(pod.Namespace).Create(pod)
 }
 
-func (k *k8sOps) GetPods(namespace string) (*v1.PodList, error) {
-	return k.getPodsWithListOptions(namespace, meta_v1.ListOptions{})
+func (k *k8sOps) GetPods(namespace string, labelSelector map[string]string) (*v1.PodList, error) {
+	return k.getPodsWithListOptions(namespace, meta_v1.ListOptions{
+		LabelSelector: mapToCSV(labelSelector),
```
Comment (piyush-nimbalkar) [inline k8s/k8s.go]: What happens to list options when the map is empty/empty list of label selectors?

File: k8s/k8s.go (hunk 1 above)
Comment (disrani-px) [reply k8s/k8s.go]: Done

File: k8s/k8s.go (hunk 2 above)
Comment (disrani-px) [reply k8s/k8s.go]: Done

**Pull request merged**